src/main/java/Parser.java

```diff
@@ -0,0 +1,125 @@
+import java.util.Scanner;
+import java.io.IOException;
+
+/**
+ * Deals with reading and making sense of the user command.
+ */
+public class Parser {
+
+    private Storage storage;
+    private TaskList tasks;
+    private Ui ui;
+
+    /**
+     * Creates a new parser instance.
+     *
+     * @param s Storage instance to be used.
+     * @param t TaskList containing task list.
+     * @param u Ui instance to be used.
+     */
+    public Parser(Storage s, TaskList t, Ui u) {
+        this.storage = s;
+        this.tasks = t;
+        this.ui = u;
```

Comment:
i think there is no need to use "this" in this situation since we can easily distinguish between 'ui' and 'u'! Same goes for 'storage' and 's', and 'tasks' and 't' :)